src/pyupgrade/plugins/unnecessary_encode_utf8.rs

```diff
@@ -0,0 +1,97 @@
+use rustpython_ast::{Constant, Expr, ExprKind, Located};
+
+use crate::{
+    ast::types::Range,
+    autofix::Fix,
+    check_ast::Checker,
+    checks::{Check, CheckKind},
+    source_code_locator::SourceCodeLocator,
+};
+
+const UTF8_LITERALS: &'static [&'static str] = &["utf-8", "utf8", "utf_8", "u8", "utf", "cp65001"];
+
+fn as_call_to_encode<'a>(func: &'a Expr, args: &Vec<Located<ExprKind>>) -> Option<&'a Box<Expr>> {
+    if let ExprKind::Attribute {
+        value: variable,
+        attr,
+        ..
+    } = &func.node
+    {
+        if attr == "encode" {
+            match args.len() {
+                0 => return Some(variable),
+                1 => match &args[0].node {
+                    ExprKind::Constant { value, .. } => match value {
+                        Constant::Str(encoding) => {
+                            if UTF8_LITERALS.contains(&encoding.to_lowercase().as_str()) {
+                                return Some(variable);
+                            }
+                        }
+                        _ => {}
+                    },
+                    _ => {}
+                },
+                _ => {}
+            }
+        }
+    }
+    None
+}
+
+fn delete_arg_check(expr: &Expr, arg: &Located<ExprKind>, patch: bool) -> Check {
+    let mut check = Check::new(CheckKind::UnnecessaryEncodeUTF8, Range::from_located(expr));
+    if patch {
+        check.amend(Fix::deletion(arg.location, arg.end_location.unwrap()));
+    }
+    check
+}
+
+fn prefix_with_b(expr: &Expr, constant: &Expr, locator: &SourceCodeLocator, patch: bool) -> Check {
+    let mut check = Check::new(CheckKind::UnnecessaryEncodeUTF8, Range::from_located(expr));
+    if patch {
+        let raw_constant = locator.slice_source_code_range(&Range {
+            location: constant.location,
+            end_location: constant.end_location.unwrap(),
+        });
+        let content = if raw_constant.starts_with("u") {
+            format!("b{}", &raw_constant[1..])
+        } else {
+            format!("b{}", raw_constant)
+        };
+        check.amend(Fix::replacement(
+            content,
+            expr.location,
+            expr.end_location.unwrap(),
+        ))
+    }
+    return check;
+}
+
+pub fn unnecessary_encode_utf8(
+    checker: &mut Checker,
+    expr: &Expr,
+    func: &Expr,
+    args: &Vec<Located<ExprKind>>,
+    locator: &SourceCodeLocator,
+) {
+    if let Some(variable) = as_call_to_encode(func, args) {
+        match &variable.node {
+            ExprKind::Constant { value, .. } => match value {
+                Constant::Str(literal) => {
+                    if literal.is_ascii() {
+                        checker.add_check(prefix_with_b(expr, variable, locator, checker.patch()));
+                    } else if let Some(arg) = args.get(0) {
```

Comment:
Should we be verifying that `args.len() == 1`, not just that it has at least one argument?